documentation/book/con-tls-within-the-cluster.adoc

```diff
@@ -0,0 +1,37 @@
+// Module included in the following assemblies:
+//
+// assembly-security.adoc
+
+[id='tls-within-the-cluster-{context}']
+= TLS within the cluster
+
+All the internal communication within the Kafka cluster is encrypted with TLS. 
+
+== Zookeeper Communication
+
+Zookeeper does not support TLS itself. 
```

Comment:
I'm trying to be more minimalist in the documentation - but if I lost the meaning just let me know.

---------------------------------
Zookeeper does not support TLS itself. 
By deploying an `stunnel` sidecar within every Zookeeper pod, the Cluster Operator is able to provide data encryption and authentication between Zookeeper nodes in a cluster.
Zookeeper communicates only with the `stunnel` sidecar over the loopback interface.
The `stunnel` sidecar then proxies all Zookeeper traffic, TLS decrypting data upon entry into a Zookeeper pod and TLS encrypting data upon departure from a Zookeeper pod.

TO:

For Zookeeper to support TLS you must deploy a `stunnel` sidecar in every Zookeeper pod. The Cluster Operator then uses to provide encryption and authentication between the Zookeeper nodes.  Zookeeper communicates only with the `stunnel` sidecar over the loopback interface and the `stunnel` sidecar then proxies all Zookeeper traffic, TLS decrypting data upon entry into a Zookeeper pod, and TLS encrypting data upon departure from a Zookeeper pod.

NOTE: Zookeeper communicates only with the `stunnel` sidecar over the loopback interface.

Reply:
"For Zookeeper to support TLS" doesn't really work for pedants because _Zookeeper doesn't support TLS_ and the stunnel thing is simply a workaround for that.

"The Cluster Operator then uses to provide encryption and authentication between the Zookeeper nodes." doesn't make sense. I assume you meant "The Cluster Operator then uses _stunnel_ to provide encryption and authentication between the Zookeeper nodes."

I'm also not convinced that the NOTE really says anything not already said.

Reply:
Yup, I had trouble with this one.  Go ahead and leave it alone for now and when @laidan6000 goes through the book later he can make it better :)